src/backend/distributed/utils/statistics_collection.c

```diff
@@ -249,10 +250,17 @@ citus_server_id(PG_FUNCTION_ARGS)
 {
 	uint8 *buf = (uint8 *) palloc(UUID_LEN);
 
+#if PG_VERSION_NUM >= 100000
 	if (!pg_backend_random((char *) buf, UUID_LEN))
 	{
 		ereport(ERROR, (errmsg("failed to generate server identifier")));
 	}
+#else
+	for (int bufIdx = 0; bufIdx < UUID_LEN; bufIdx++)
+	{
+		buf[bufIdx] = random() & 0xff;
```

Comment:
Can you add an explicit cast to `char` here: `(char) (random() & 0xFF)`?

The non-strong-random implementation of `pg_backend_random` looks like so:

```c
	for (i = 0; dst < end; i++)
	{
		uint32		r;
		int			j;

		/*
		 * pg_jrand48 returns a 32-bit integer. Fill the next 4 bytes from it.
		 */
		r = (uint32) pg_jrand48(BackendRandomShmem->seed);

		for (j = 0; j < 4 && dst < end; j++)
		{
			*(dst++) = (char) (r & 0xFF);
			r >>= 8;
		}
	}
```

So for consistency, I think we should have the cast and `FF` be uppercase, but otherwise this is good.